src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +592,27 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static makeIdQuery(idOrIds) {
+    const idIsValid = mongoose.Types.ObjectId.isValid;
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.");
+      }
+
+      idQuery = idOrIds;
+    } else if (Array.isArray(idOrIds)) {
+      let validIds = idOrIds.filter(idIsValid);
```

Comment:
400 just `filter`/`labelMapper` ones; `/collection/:id` could still 404. That is, something like:

```
if (typeof idOrIds === "string") {
   if (!idIsValid(idOrIds)) {
     throw new APIError(404, undefined, "No matching resource found.");
   }

   idQuery = idOrIds;
}

else if (Array.isArray(idOrIds)) {
  if(!idOrIds.every(idIsValid)) {
     throw new APIError(400, undefined, "One or more of the provided ids was invalid");
  }
  // etc
}
```